### DISCUSSION TITLE
[Feature Request] Call SpeedDialChild.onTap directly if only one child

### DISCUSSION THREAD

**Discussion by mtc-jed** [darioielardi/flutter_speed_dial#237] [Ideas]
If there is only one child provided to `SpeedDial.children` and a new boolean `SpeedDial.callOnTapDirectlyIfOnlyOneChild` is true, call the only child's `onTap` without opening the Speed Dial.
This would allow for much cleaner code when the available `SpeedDialChild`s are restricted in one way or another since we wouldn't have to calculate the conditions twice. I don't think having a speed dial with only one child is very nice.
For now I'll make my own widget to handle this.

My use case is certain clients have certain options available in the app, but others are of no use to them. Hence they are hidden.

Edit : Might as well put that code out there :
```dart
class SpeedDialPlus extends StatelessWidget {

  final List<SpeedDialChild> children;
  
  const SpeedDialPlus({required this.children, Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    
    if(children.isEmpty) return Container();
    
    if(children.onlyOne) return FloatingActionButton(
      onPressed: children.first.onTap,
      child: const Icon(Icons.add_rounded),
    );
    
    return SpeedDial(
      icon: Icons.add_rounded,
      activeIcon: Icons.close,
      spacing: 4,
      iconTheme: const IconThemeData(color: Colors.white),
      children: children,
    );
  }
}
```

**Comment by luca-colazzo** [comment]
You can open a PR to submit your enhancements
